PWGJE/Tasks/PhotonIsolationQA.cxx

```diff
@@ -0,0 +1,364 @@
+    MC_Info.add("hEnergy_ShowerShapeShort", "Energy vs Shower shape short axis", o2HistType::kTH2F, {{Energy_Axis}, {Shower_Shape_Short_Axis}});
+    MC_Info.add("hEnergy_NCells", "Energy vs Number of cells in cluster", o2HistType::kTH2F, {{Energy_Axis}, {NCells_Axis}});
+
+    MC_Info.add("hEvsNumTracks", "Energy of cluster vs matched tracks", o2HistType::kTH2F, {{Energy_Axis}, {Num_Track_Axis}});
+    MC_Info.add("hTrackPt", "Pt of matched track", o2HistType::kTH1F, {Pt_Axis});
+    MC_Info.add("hTrackEta", "Eta of matched track", o2HistType::kTH1F, {Eta_Axis});
+    MC_Info.add("hTrackPhi", "Phi of matched track", o2HistType::kTH1F, {Phi_Axis});
+    MC_Info.add("hRatioClusterETrackP", "Ratio between energy of cluster and P of tracks", o2HistType::kTH1F, {RatioP_Axis});
+    MC_Info.add("hRatioClusterETrackPrackPt", "Ratio between E and P vs Pt of matched tracks", o2HistType::kTH2F, {{RatioP_Axis}, {Pt_Axis}});
+
+    MC_Info.add("hEvsPtIso", "Pt_Iso", o2HistType::kTH2F, {{Energy_Axis}, {PtIso_Axis}});
+    MC_Info.add("hRho_Perpen_Cone", "Density of perpendicular cone", o2HistType::kTH1F, {Rho_Axis});
+    MC_Info.add("hShowerShape", "Shower shape", o2HistType::kTH2F, {{Shower_Shape_Long_Axis}, {Shower_Shape_Short_Axis}});
+    MC_Info.add("hSigmaLongvsPtIso", "Long shower shape vs Pt_Iso", o2HistType::kTH2F, {{Shower_Shape_Long_Axis}, {PtIso_Axis}});
+    MC_Info.add("hABCDControlRegion", "Yield Control Regions", o2HistType::kTH2F, {{ABCD_Axis}, {Energy_Axis}});
+    MC_Info.add("hMcParticlesToCluster", "Energy of particles linked to mc cluster", o2HistType::kTH1F, {{100, 0, 20}});
+    MC_Info.add("hMcParticleStatusCode", "status code of mc particle linked to mc cluster", o2HistType::kTH1F, {{100, 0, 100}});
+
+    std::vector<std::string> bin_names = {"A", "B", "C", "D", "True Bckgr A"};
+    for (int i = 0; i < bin_names.size(); i++) {
+      MC_Info.get<TH2>(HIST("hABCDControlRegion"))->GetXaxis()->SetBinLabel(i + 1, bin_names[i].c_str());
+      Data_Info.get<TH2>(HIST("hABCDControlRegion"))->GetXaxis()->SetBinLabel(i + 1, bin_names[i].c_str());
+    }
+  }
+
+  // boolian returns true if a track is matched to the cluster with E_Cluster/P_track < 1.75. Otherwise returns false
+  bool track_matching(const auto& cluster, o2::aod::EMCALMatchedTracks const& matched_tracks)
+  {
+    for (const auto& match : matched_tracks) {
+      double dEta = abs(match.track_as<myGlobTracks>().trackEtaEmcal() - cluster.eta());
+      double dPhi = match.track_as<myGlobTracks>().trackPhiEmcal() - cluster.phi();
+      if (dPhi < 0) {
+        dPhi = 2. * M_PI - dPhi;
+      }
+      double distance = sqrt(pow(dEta, 2) + pow(dPhi, 2));
+      if (distance <= TrackMatchingRadius) {
+        double abs_p = abs(match.track_as<myGlobTracks>().p());
+        if ((cluster.energy() / abs_p) < 1.75) {
+          return true;
+        }
+      }
```

Comment:
I am wondering how this method was developed. In run2 we use a pt-dependent matching with a matching function, independent in eta and phi. Having a hard coded, and rather large cut, can lead to fake matches at high-E.

Reply:
@mfasDa for Run 2 prompt photons we usually used a fixed dEta and dPhi.  We choose a wide cut, because we found that missed matches may lead to unnatural shapes in the isolation energy that are self correlated with photon pt. In any case, for isolated prompt photons the track matching has strong interplay. This the track matching misses get caught by isolation cut and vice versa